src/plugins/data_explorer/public/components/sidebar/index.tsx

```diff
@@ -4,20 +4,36 @@
  */
 
 import React, { FC, useCallback, useEffect, useState } from 'react';
-import { EuiPageSideBar, EuiSplitPanel } from '@elastic/eui';
+import {
+  EuiPageSideBar,
+  EuiSplitPanel,
+  EuiModal,
+  EuiModalHeader,
+  EuiModalBody,
+  EuiModalHeaderTitle,
+  EuiModalFooter,
+  EuiButtonEmpty,
+  EuiButton,
+  EuiText,
+} from '@elastic/eui';
 import { i18n } from '@osd/i18n';
 import { DataSourceGroup, DataSourceSelectable, DataSourceType } from '../../../../data/public';
 import { DataSourceOption } from '../../../../data/public/';
 import { useOpenSearchDashboards } from '../../../../opensearch_dashboards_react/public';
 import { DataExplorerServices } from '../../types';
 import { setIndexPattern, useTypedDispatch, useTypedSelector } from '../../utils/state_management';
 
+const LOG_EXPLORER_TITLE_HINT = ' - Opens in Log Explorer';
+const DATA_SOURCE_SELECTOR_CONFIGS = { customGroupTitleExtension: LOG_EXPLORER_TITLE_HINT };
+
 export const Sidebar: FC = ({ children }) => {
   const { indexPattern: indexPatternId } = useTypedSelector((state) => state.metadata);
   const dispatch = useTypedDispatch();
   const [selectedSources, setSelectedSources] = useState<DataSourceOption[]>([]);
   const [dataSourceOptionList, setDataSourceOptionList] = useState<DataSourceGroup[]>([]);
   const [activeDataSources, setActiveDataSources] = useState<DataSourceType[]>([]);
+  const [modalContent, setModalContent] = useState<React.ReactNode | null>(null);
+  const [isModalVisible, setIsModalVisible] = useState(false);
```

Comment:
Cant we simply use the fact that the modal content is null to open and close the modal?

Reply:
make sense, removed